src/__tests__/rate-limiter.test.ts

```diff
@@ -85,18 +85,18 @@ describe('Rate Limiter', () => {
 
         it('rate limits when past the threshold ', () => {
             range(100).forEach(() => {
-                expect(rateLimiter.isCaptureClientSideRateLimited()).toBe(false)
+                expect(rateLimiter.isCaptureClientSideRateLimited().isRateLimited).toBe(false)
             })
             range(200).forEach(() => {
-                expect(rateLimiter.isCaptureClientSideRateLimited()).toBe(true)
+                expect(rateLimiter.isCaptureClientSideRateLimited().isRateLimited).toBe(true)
             })
             expect(persistedBucket['$capture_rate_limit']).toEqual({
                 tokens: 0,
                 last: systemTime,
             })
 
             moveTimeForward(2000) // 2 seconds = 20 tokens
-            expect(rateLimiter.isCaptureClientSideRateLimited()).toBe(false)
+            expect(rateLimiter.isCaptureClientSideRateLimited().isRateLimited).toBe(false)
```

Comment:
```suggestion
            expect(rateLimiter.clientRateLimitContext().isRateLimited).toBe(false)
```

it's much clearer in the implementation because we assign this to `clientRateLimitContext`

`isCaptureClientSideRateLimited().isRateLimited` is a strange repetition that confused me hitting this file in the review first